R/steps_and_checks.R

```diff
@@ -51,3 +68,22 @@ add_check <- function(rec, object) {
   rec$steps[[length(rec$steps) + 1]] <- object
   rec
 }
+
+# ------------------------------------------------------------------------------
+
+check_step_check_args <- function(x, call = rlang::caller_env()) {
```

Comment:
i like this idea! makes it very clean. what about checking for argument used a lot but not everywhere. like `keep_original_cols`?